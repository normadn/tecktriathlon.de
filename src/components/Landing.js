import React from "react";

import Header from "./Header";
import ActionButton from "./ActionButton";
import ContentWrapper from "./ContentWrapper";

import landingImg from "../assets/img/logo/tecktriathlon-hero-new.svg";
import logo from "../assets/img/logo/logo-no-text.svg";
import word from "../assets/img/logo/tecktriathlon.svg";
import backgroundCubes from "../assets/img/backgrounds/backgroundCubes.svg";

const applyButtonCSS = {
  margin: "5px 0 30px 0",
  "@media(max-width: 820px)": {
    margin: "5px 0 15px auto"
  }
};

const Landing = () => (
  <div
    id=""
    css={{
      width: "100%",
      backgroundSize: "cover",
      "@media(max-width: 600px)": {
        minHeight: "600px"
      }
    }}
  >
    <Header standaloneVersion={false} />
    <ContentWrapper>
      <section
        css={{
          display: "grid",
          gridTemplateColumns: "auto 1fr",
          marginBottom: "20px",
          "@media(max-width: 820px)": {
            textAlign: "center",
            display: "block"
          }
        }}
      >
        <div
          css={{
            "@media(min-width: 1200px)": {
              paddingRight: "50px",
              paddingTop: "90px"
            },
            "@media (min-width: 821px) and (max-width: 1199px)": {
              paddingRight: "50px",
              paddingTop: "75px"
            },
            "@media (max-width: 820px)": {
              paddingTop: "80px"
            }
          }}
        >
          <img
            src={landingImg}
            css={{
              width: "70vw",
              maxWidth: "300px",
              "@media(min-width: 821px)": { display: "none" }
            }}
            alt="Illustration"
          />
          <br />
          <h1
            css={{
              paddingTop: "40px",
              paddingBottom: "18px",
              fontSize: "40px",
              "@media(max-width: 820px)": {
                paddingTop: "12px",
                paddingBottom: "8px"
              }
            }}
          >
            Teck Triathlon
          </h1>
          <h2
            data-cy="qhacks-information"
            css={{
              color: "#020066",
              fontWeight: 600,
              fontSize: "22px",
              paddingBottom: "30px",
              "@media(max-width: 820px)": {
                paddingBottom: "24px"
              }
            }}
          >
            Volkstriathlon in Bissingen an der Teck
          </h2>

          <h2
            data-cy="qhacks-information"
            css={{
              color: "#575757",
              fontWeight: 600,
              fontSize: "20px",
              paddingBottom: "20px",
              "@media(max-width: 820px)": {
                paddingBottom: "24px"
              }
            }}
          >
            Es ist egal wie schnell du bist.
            <br />
            Du schl√§gst alle, die zuhause bleiben!
          </h2>
          {/* <div>
            <ActionButton
              backgroundColor="#ffffff"
              foregroundColor="#00205b"
              style={applyButtonCSS}
              link="https://drive.google.com/file/d/1syJuMCO71SGx7nsQcDAR6qNrpehHjuLJ/view?usp=sharing"
              type="rounded"
            >
              Ergebnisse 2023
            </ActionButton>
          </div> */}
          <h2
            css={{
              padding: "20px 0 30px",
              color: "#575757",
              fontWeight: 800,
              lineHeight: "30px",
              "@media(max-width: 820px)": {
                paddingTop: "10px",
                fontSize: "15px",
                lineHeight: "1.6"
              }
            }}
          >
            Termin: 31.08.2024
          </h2>
          {/* <h2
            css={{
              color: "#575757",
              fontWeight: 600,
              fontSize: "20px",
              paddingBottom: "30px",
              "@media(max-width: 820px)": {
                paddingBottom: "24px"
              }
            }}
          >
            Wir sagen Danke!
          </h2> */}
          {/* <div>
            <ActionButton
              backgroundColor="#ffffff"
              foregroundColor="#00205b"
              style={applyButtonCSS}
              link="https://forms.gle/p6Ewhf3p2ABaeBiJ9"
              type="rounded"
            >
              Warteliste
            </ActionButton>
          </div> */}
          <div>
            <ActionButton
              backgroundColor="#ffffff"
              foregroundColor="#00205b"
              style={applyButtonCSS}
              link="https://forms.gle/Uy27qiQ98PfYuEnw7"
              type="rounded"
            >
              Einzelanmeldung
            </ActionButton>
          </div>
          <div>
            <ActionButton
              backgroundColor="#ffffff"
              foregroundColor="#00205b"
              style={applyButtonCSS}
              link="https://forms.gle/SkjqfVHgdZtkuGBk6"
              type="rounded"
            >
              Teamanmeldung
            </ActionButton>
          </div>
          {/* <div>
            <ActionButton
              backgroundColor="#ffffff"
              foregroundColor="#00205b"
              style={applyButtonCSS}
              link="https://forms.gle/nyrtwfaRL6jfWP7m8"
              type="rounded"
            >
              Interesse Teck Triathlon 2024
            </ActionButton>
          </div> */}
        </div>
        <div
          css={{
            position: "relative",
            textAlign: "right",
            display: "flex",
            paddingTop: "85px",
            alignItems: "right",
            "@media(max-width: 820px)": {
              display: "none"
            }
          }}
        >
          <img
            src={landingImg}
            css={{
              paddingTop: "35px",
              paddingLeft: "50px",
              paddingBottom: "20px",
              width: "90%"
            }}
            alt="TVB logo"
          />
        </div>
      </section>
    </ContentWrapper>
  </div>
);

export default Landing;
