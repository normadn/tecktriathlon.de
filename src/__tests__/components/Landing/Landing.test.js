import React from "react";
import Landing from "../../../components/Landing";
import { shallow } from "enzyme";

describe("Landing", () => {
  it("renders correctly", () => {
    expect(shallow(<Landing />)).toMatchSnapshot();
  });
});
