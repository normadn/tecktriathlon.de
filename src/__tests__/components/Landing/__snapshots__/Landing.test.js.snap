// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing renders correctly 1`] = `
<div
  css={
    Object {
      "@media(max-width: 600px)": Object {
        "minHeight": "600px",
      },
      "background": "url(test-file-stub) no-repeat center center",
      "backgroundSize": "cover",
      "width": "100%",
    }
  }
  id=""
>
  <Header
    standaloneVersion={false}
  />
  <_default>
    <section
      css={
        Object {
          "@media(max-width: 820px)": Object {
            "display": "block",
            "textAlign": "center",
          },
          "display": "grid",
          "gridTemplateColumns": "auto 1fr",
        }
      }
    >
      <div
        css={
          Object {
            "@media (max-width: 820px)": Object {
              "paddingTop": "80px",
            },
            "@media (min-width: 821px) and (max-width: 1199px)": Object {
              "paddingRight": "50px",
              "paddingTop": "215px",
            },
            "@media(min-width: 1200px)": Object {
              "paddingRight": "50px",
              "paddingTop": "215px",
            },
          }
        }
      >
        <img
          alt="QHacks Crown"
          css={
            Object {
              "@media(max-width: 820px)": Object {
                "display": "none",
              },
            }
          }
          data-cy="qhacks-crown"
          src="test-file-stub"
          width="170px"
        />
        <img
          alt="QHacks Illustration"
          css={
            Object {
              "@media(min-width: 821px)": Object {
                "display": "none",
              },
              "maxWidth": "350px",
              "width": "75vw",
            }
          }
          src="test-file-stub"
        />
        <br />
        <img
          alt="QHacks Wordmark"
          css={
            Object {
              "@media(max-width: 820px)": Object {
                "paddingBottom": "8px",
                "paddingTop": "12px",
                "width": "192px",
              },
              "maxWidth": "300px",
              "paddingBottom": "18px",
              "paddingTop": "40px",
              "width": "100vw",
            }
          }
          data-cy="qhacks-wordmark"
          src="test-file-stub"
        />
        <h2
          css={
            Object {
              "@media(max-width: 820px)": Object {
                "paddingBottom": "24px",
              },
              "color": "#00205b",
              "fontSize": "22px",
              "fontWeight": 600,
              "paddingBottom": "30px",
            }
          }
          data-cy="qhacks-information"
        >
          Queen’s University • February 1-3, 2019
        </h2>
        <h2
          css={
            Object {
              "@media(max-width: 820px)": Object {
                "fontSize": "15px",
                "lineHeight": "1.6",
                "paddingTop": "20px",
              },
              "color": "#575757",
              "fontWeight": 500,
              "lineHeight": "30px",
              "padding": "20px 0 100px",
            }
          }
          data-cy="qhacks-tagline"
        >
          Queen’s University’s annual hackathon.
          <br />
          Join us for an unforgettable experience!
        </h2>
      </div>
      <div
        css={
          Object {
            "@media(max-width: 820px)": Object {
              "display": "none",
            },
            "alignItems": "center",
            "display": "flex",
            "paddingTop": "84px",
            "position": "relative",
            "textAlign": "right",
          }
        }
      >
        <img
          alt="QHacks Illustration"
          css={
            Object {
              "width": "100%",
            }
          }
          src="test-file-stub"
        />
      </div>
    </section>
  </_default>
</div>
`;
